src/transform-sdk/cpp/include/redpanda/transform_sdk.h

```diff
@@ -25,6 +26,26 @@
 #include <system_error>
 #include <vector>
 
+namespace detail {
+
+template<typename T, typename Tag>
+struct simple_named_type {
+    using type = T;
+    explicit simple_named_type(type val)
+      : _val(val) {}
+    type operator()() const { return _val; }
```

Comment:
Can we use an explicit `value` here? I am personally not a fan of the call operator to access the underlying data.

Reply:
yeah sure. I don't have a strong preference

Reply:
done